skbio/io/fasta.py

```diff
@@ -557,6 +563,10 @@
 
 @register_sniffer('fasta')
 def _fasta_sniffer(fh):
+
+    if _count_blank_lines(fh) > 5:
```

Comment:
Agree we should be able to push line skipping into `_line_generator`. So the way it would work is if `skip_blanks` is `True` it will consume (up to `max_blanks` if set, raising an exception if exceeded) blanks at the beginning of the file. Once the first nonblank line is read, blanks of ignored by `_line_generator` and handled by the caller. Seem reasonable?